obspy/clients/fdsn/mass_downloader/__init__.py

```diff
@@ -0,0 +1,503 @@
+Instances of these classes will later be passed to the function sparking the
+downloading process. A rectangular domain for example is defined like this:
+
+>>> from obspy.clients.fdsn.mass_downloader.domain import RectangularDomain
+>>> domain = RectangularDomain(minlatitude=-10, maxlatitude=10,
+...                            minlongitude=-10, maxlongitude=10)
+
+Additional restrictions like temporal bounds, SEED identifier wildcards,
+and other things are set with the help of
+the :class:`~obspy.clients.fdsn.mass_downloader.restrictions.Restrictions`
+class. Please refer to its documentation for a more detailed explanation of
+the parameters.
+
+>>> from obspy import UTCDateTime
+>>> from obspy.clients.fdsn.mass_downloader import Restrictions
+>>> restrict = Restrictions(
+...     starttime=UTCDateTime(2012, 1, 1),
+...     endtime=UTCDateTime(2012, 1, 1, 1),
+...     network=None, station=None, location=None, channel=None,
+...     reject_channels_with_gaps=True,
+...     minimum_length=0.9,
+...     minimum_interstation_distance_in_m=1000,
+...     channel_priorities=["HH[Z,N,E]", "BH[Z,N,E]"],
+...     location_priorities=["", "00", "01"])
+
+
+Step 2: Storage Options
+~~~~~~~~~~~~~~~~~~~~~~~
+
+After determining what to download, the helpers must know where to store the to
+be downloaded data. This requires some flexibility in case this is integrated
+as a component into a bigger system. An example of this is a toolbox that has a
+database to manage its data. A major concern is to not download already
+existing data. In order to enable such a use case the download helpers can be
+given functions that are evaluated when determining the filenames of the to be
+downloaded data.  Depending on the return value, the helper class will download
+the whole, only parts, or even nothing, of that particular piece of data.
+
+Storing MiniSEED waveforms
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The MiniSEED storage rules are set by the ``mseed_storage`` argument of the
+:meth:`~obspy.clients.fdsn.mass_downloader.mass_downloader.MassDownloader.download`
+method of the
+:class:`~obspy.clients.fdsn.mass_downloader.mass_downloader.MassDownloader`
+class
+
+
+**Option 1: Folder Name**
+
+In the simplest case it is just a folder name:
+
+>>> mseed_storage = "waveforms"
+
+This will cause all MiniSEED files to be stored as
+
+``waveforms/NETWORK.STATION.LOCATION.CHANNEL__STARTTIME__ENDTIME.mseed``.
+
+An example of this is
+
+``waveforms/BW.FURT..BHZ__2014-10-27T16-37-23Z__2014-10-27T16-37-33Z.mseed``
+
+which is rather general but also quite long.
+
+**Option 2: String Template**
+
+For more control use the second possibility and provide a string containing
+``{network}``, ``{station}``, ``{location}``, ``{channel}``, ``{starttime}``,
+and ``{endtime}`` format specifiers. These values will be interpolated to
+acquire the final filename. The start and end times will be formatted with
+``strftime()`` with the specifier ``'%Y-%m-%dT%H-%M-%SZ'`` in an effort to
```

Comment:
Is it really necessary to use the dashes? ISO8601 allows you to drop the colons.